### ISSUE TITLE
Moving to work with promises instead of callbacks ?

### ISSUE THREAD

**Issue by chenfli** [pkgcloud/pkgcloud#537]
Hi,
Is using promises is on the road map ?
Using callbacks forces me to do up to 5 level of indentation.
I love your work and hope to see promises usage in the coming versions.

Thanks
Chen

**Comment by jamespedid** [comment]
You can still work with Promises; just create them yourself and resolve in the callback you provide.

**Comment by chenfli** [comment]
@jamespedid you are right amd I did managed to use it with blubird. It's just that it a bit cumbersome. Since you are not prototype all the components it means that I need to promisofyAll per component, for each module that I use, quite redundent for my opinion.

And another question, are you going to use script type in the coming versions ?

**Comment by fcanela** [comment]
:+1: for promise support. Native promises and async/await is available in Node.js and callbacks are less used for asynchronous handling. If someone considers developing it at some point know that there are users interested.

Thanks for the awesome job with the package.

**Comment by DaniGuardiola** [comment]
Yeah this would be an awesome addition!
